vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java

```diff
@@ -96,9 +97,20 @@
     public void initVariableUsingTemplate(Set<VariableScope> scopes, String variableName, String templatePath,
             ExamplesTable templateParameters) throws IOException, TemplateException
     {
-        Map<String, ?> dataModel = MapUtils.convertExamplesTableToMap(templateParameters);
-        String value = freemarkerProcessor.process(templatePath, Map.of("parameters", dataModel),
-                StandardCharsets.UTF_8);
+        Map<String, List<String>> dataModel = MapUtils.convertExamplesTableToMap(templateParameters);
+        String value;
+        Set<String> keySet = dataModel.keySet();
+        Optional<String> firstKey = keySet.stream().findFirst();
+        if (keySet.size() == 1 && dataModel.get(firstKey.orElse(null)).size() == 1)
+        {
+            String key = firstKey.get();
+            value = freemarkerProcessor
+                    .process(templatePath, Map.of(key, dataModel.get(key).get(0)), StandardCharsets.UTF_8);
+        }
+        else
+        {
+            value = freemarkerProcessor.process(templatePath, Map.of("parameters", dataModel), StandardCharsets.UTF_8);
```

Comment:
create a single map containing both multidimensional and simple parameters and pass it to the template